src/index.ts

```diff
@@ -89,60 +90,105 @@ export default async function (manager: Manager) {
   manager.addEventListener('pageview', onEvent(true))
   manager.addEventListener('event', onEvent())
 
-  manager.registerEmbed('post', async ({ parameters }) => {
+  manager.registerEmbed('post', async ({ parameters, client }) => {
     const tweetId = parameters['tweet-id']
     const randomToken = [...Array(11)]
       .map(() => (Math.random() * 36).toString(36)[2])
       .join('')
+    const userAgent = client.userAgent
     const tweetResponse = await manager.useCache(
       'tweet_' + tweetId,
       async () => {
-        const res = await fetch(
-          `https://cdn.syndication.twimg.com/tweet-result?id=${tweetId}&token=${randomToken}`
+        const res = await manager.fetch(
+          `https://cdn.syndication.twimg.com/tweet-result?id=${tweetId}&token=${randomToken}`,
+          {
+            headers: {
+              accept:
+                'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7',
+              'User-Agent': `${userAgent}`,
+            },
+          }
         )
-        return await res.json()
+        if (!res) {
+          throw new Error('Failed to fetch tweet data.')
+        }
+        const responseString = await res.json()
+        return JSON.stringify(responseString)
       },
       600 // Cache the Tweet for 10 minutes
     )
 
-    const { user, created_at } = tweetResponse
+    const {
+      user,
+      created_at,
+      text,
+      favorite_count,
+      conversation_count,
+      quoted_tweet,
+    } = JSON.parse(tweetResponse)
     const dt = new Date(created_at)
 
-    const datetime =
-      dt.toLocaleString('en-US', {
-        hour: 'numeric',
-        minute: 'numeric',
-        hour12: true,
-      }) +
-      ' · ' +
-      dt.toLocaleString('en-US', { month: 'short' }) +
-      ' ' +
-      dt.getDate() +
-      ', ' +
-      dt.getFullYear()
-
-    let profileImage = manager.get('profileImage_' + user.screen_name)
+    const datetime = `${dt.toLocaleString('en-US', {
+      hour: 'numeric',
+      minute: 'numeric',
+      hour12: true,
+    })} · ${dt.toLocaleString('en-US', {
+      month: 'short',
+    })} ${dt.getDate()}, ${dt.getFullYear()}`
+
+    let profileImage = await manager.get('profileImage_' + user.screen_name)
 
     if (!profileImage) {
-      const res = await fetch(user.profile_image_url_https)
-      profileImage = base64Encode(await res.arrayBuffer())
-      manager.set('profileImage_' + user.screen_name, profileImage)
+      const res = await manager.fetch(user.profile_image_url_https, {
+        headers: {
+          Accept: 'image/jpeg,image/png,image/*,*/*;q=0.8',
+          'User-Agent': `${userAgent}`,
+        },
+        method: 'GET',
+      })
+      if (!res || !res.ok) {
+        throw new Error(
```

Comment:
curious as to how big this blast radius is when it happens? do other Twitter MC actions still fire?

Reply:
I removed it.